remove `pnpm format` in pre-commit hook

chenxin-yan opened the pull request:
format check is already handled by `lint-staged` (configured in `pacakge.json`). It is more efficient because it only check for changes not the entire codebase.

package.json:
```json
...
  "lint-staged": {
    "*.{js,jsx,ts,tsx}": [
      "next lint",
      "prettier --check"
    ],
    "*.{json,yml,yaml,md}": [
      "prettier --check"
    ]
  },
...
```

safipatel commented:
@chenxin-yan 
that's fine with me, but idk if you see this too: it doesn't seem to fix or prevent a commit if I have a badly formated file, thats why I had put the pnpm format in the first place, but if you have a way of doing that but in the above way that's best I guess:

Test it out and see if you're also prevented from commiting/auto fix the format:
![image](https://github.com/user-attachments/assets/3b43e298-2f40-4111-8658-f2800f5eea10)

![image](https://github.com/user-attachments/assets/f6a0e203-dfe9-4845-939f-dc1b7eeb7046)

chenxin-yan commented:
Got it! I think you're right. I reverted the change and removed the formatter from the `lint-staged` config. For the command in `.husky`, did you intentionally add `pnpm format`, or would `pnpm format:check` make more sense? The latter won't change the code and will only provide warnings. It's about preference, but it makes sure git hook does not make changes that we are not aware of for formatting.

safipatel commented:
The real issue was that the .lintstagerc was overriding our package.json configuration. So I put the prettier stuff into the .lintstagerc and it works fine now. I agree checking is better than writing. We should also probably keep an eye on this and whether this is helping or annoying people more, just cause sometimes people want to push broken code for others to checkout.

chenxin-yan commented:
> The real issue was that the .lintstagerc was overriding our package.json configuration. So I put the prettier stuff into the .lintstagerc and it works fine now. I agree checking is better than writing. We should also probably keep an eye on this and whether this is helping or annoying people more, just cause sometimes people want to push broken code for others to checkout.

You can use `git commit -m "<your message>" --no-verify` to skip pre-commit hook.
